hazelcast/src/test/java/com/hazelcast/jet/core/metrics/JobLifecycleMetricsTest.java

```diff
@@ -150,23 +158,27 @@ public void jobRestarted() {
     @Test
     public void jobCancelled() {
         //init
-        Job job = hzInstances[0].getJet().newJob(streamingPipeline());
+        Job job = hzInstances[0].getJet().newJob(streamingPipeline(),
+                new JobConfig().setStoreMetricsAfterJobCompletion(true));
         assertJobStatusEventually(job, RUNNING);
 
         assertTrueEventually(() -> assertJobStats(1, 1, 0, 0, 0));
         assertTrueAllTheTime(() -> assertJobStats(1, 1, 0, 0, 0), 1);
 
         //when
         job.cancel();
+        assertTrueEventually(() -> assertJobStatusMetric(job, FAILED));
```

Comment:
I am afraid that we might get request to add `isUserCancelled` flag to the metrics as well, but if so, we can do this separately

Reply:
https://hazelcast.atlassian.net/browse/HZ-3610

Reply:
3e5c055e86a824a4d0b6efa83f56fe9da60d6223